[Shell] Crash when switching Bottom Navigation Tabs too fast

### Description
```
11-12 14:14:39.915 E/mono    ( 4794): Unhandled Exception:
11-12 14:14:39.915 E/mono    ( 4794): System.ObjectDisposedException: Cannot access a disposed object.
11-12 14:14:39.915 E/mono    ( 4794): Object name: 'Xamarin.Forms.Platform.Android.ColorChangeRevealDrawable'.
11-12 14:14:39.915 E/mono    ( 4794):   at (wrapper dynamic-method) System.Object.25(intptr,intptr,intptr)
11-12 14:14:39.915 E/mono-rt ( 4794): [ERROR] FATAL UNHANDLED EXCEPTION: System.ObjectDisposedException: Cannot access a disposed object.
11-12 14:14:39.915 E/mono-rt ( 4794): Object name: 'Xamarin.Forms.Platform.Android.ColorChangeRevealDrawable'.
11-12 14:14:39.915 E/mono-rt ( 4794):   at (wrapper dynamic-method) System.Object.25(intptr,intptr,intptr)
```

### Steps to Reproduce

1.  pull down shell-3.4.0
2. Run the Shell Control Gallery project 
3. click between the bottom tabs quickly

### Expected Behavior
Nothing should crash

### Actual Behavior
It Crashes

### Basic Information

- Version with issue:
- Last known good version:
- IDE:
- Platform Target Frameworks: <!-- All that apply -->
  - Android: 8.1
- Android Support Library Version: v27 and v28

--- a/Xamarin.Forms.Platform.Android/Renderers/ColorChangeRevealDrawable.cs
+++ b/Xamarin.Forms.Platform.Android/Renderers/ColorChangeRevealDrawable.cs
@@ -60,7 +60,8 @@ namespace Xamarin.Forms.Platform.Android
 		void OnUpdate(object sender, ValueAnimator.AnimatorUpdateEventArgs e)
 		{
 			_progress = (float)e.Animation.AnimatedValue;
-			InvalidateSelf();
+			if (!this.IsDisposed())
+				InvalidateSelf();
 		}
 	}
 }

--- a/Xamarin.Forms.Platform.Android/Renderers/ShellItemRenderer.cs
+++ b/Xamarin.Forms.Platform.Android/Renderers/ShellItemRenderer.cs
@@ -138,7 +138,8 @@ namespace Xamarin.Forms.Platform.Android
 					void clickCallback(object s, EventArgs e)
 					{
 						selectCallback(shellContent, bottomSheetDialog);
-						innerLayout.Click -= clickCallback;
+						if (!innerLayout.IsDisposed())
+							innerLayout.Click -= clickCallback;
 					}
 					innerLayout.Click += clickCallback;
 

